WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -149,6 +149,10 @@ public async Task<bool> StartRoundAsync(IEnumerable<SmartCoin> smartCoins, Round
 			}
 			finally
 			{
+				foreach (var aliceClient in registeredAliceClients)
+				{
+					aliceClient.Finish();
```

Comment:
why not a `Disposable`? is it because having a collection of disposables requires some wrapper type that is itself disposable?

Reply:
`AliceClient` is not disposable because it doesn't hold any unmanaged resource. OTOH only `IDisposable` classes are expected to have a `Dispose` method, the rest have `Release`, `Close`, `Finish`, `Stop` and similars.

Finally, the disposable pattern is very complex (at the point that nobody implements it correctly) and requires a lot of plumbing and a lot of ceremonies and ritual dancing that we don't need/want here.